hoomd/pytest/test_simulation.py

```diff
@@ -97,11 +101,21 @@ def test_initialization(device, simulation_factory, get_snapshot):
     sim.operations.schedule()
     sim.run(1)
 
+    assert sim.always_compute_pressure is False
+    sim.always_compute_pressure = True
+    assert sim.always_compute_pressure is True
 
-def test_run(simulation_factory, get_snapshot):
-    sim = simulation_factory(get_snapshot())
-    sim.operations.schedule()
+
+def test_run(simulation_factory, get_snapshot, device):
+    sim = hoomd.simulation.Simulation(device)
+    assert sim.timestep is None
+    sim.timestep = 10
+    sim = simulation_factory(get_snapshot())  # set state
     assert sim.timestep == 0
```

Comment:
Shouldn't this fail now? Since we are setting the timestep to 10?

Reply:
I believe `timestep` is being set to 0 once `sim.state` is set. I'm not sure if this is the expected behavior

Reply:
This is not. Must happen in the construction of the `System` object.

Reply:
The sim object is being reset to a new simulation by `simulation_factory`, overriding the previous object.
```
    sim = hoomd.simulation.Simulation(device)
    sim.timestep = 10
    sim = simulation_factory(get_snapshot())  # set state
    assert sim.timestep == 0
```
I think you mean:
```
    sim = hoomd.simulation.Simulation(device)
    sim.timestep = 10
    sim.create_from_snapshot(get_snapshot())
    assert sim.timestep == 0
```

Reply:
`sim.timestep` is still being set to 0 when I use `create_state_from_snapshot`. I believe this is because `sim.timestep` returns None until `_cpp_sys` is set, while `sim._timestep` returns what it is set to (in this case 10).

If I change:
```
def create_state_from_snapshot(self, snapshot):
        # initialize the system
        # Error checking
        if self.state is not None:
            raise RuntimeError("Cannot initialize more than once\n")

        self._state = State(self, snapshot)

        step = 0
        if self.timestep is not None:
            step = self.timestep

        # Store System and Reader for Operations
        self._cpp_sys = _hoomd.System(self.state._cpp_sys_def, step)
        self._init_communicator()
        self._cpp_sys.enableQuietRun(not self.verbose_run)
```

to
```
def create_state_from_snapshot(self, snapshot):
        # initialize the system
        # Error checking
        if self.state is not None:
            raise RuntimeError("Cannot initialize more than once\n")

        self._state = State(self, snapshot)

        step = 0
        if self._timestep is not None:
            step = self._timestep

        # Store System and Reader for Operations
        self._cpp_sys = _hoomd.System(self.state._cpp_sys_def, step)
        self._init_communicator()
        self._cpp_sys.enableQuietRun(not self.verbose_run)
```

and set `self._timestep` to None in `__init__`, the timestep gets set properly.

Reply:
Then we should fix the `timestep` property. If the user sets it, it should no longer return `None`, but the value that the user set.

Reply:
Fixed 7f0d69ad9